agent/command.go

```diff
@@ -147,6 +147,9 @@ func (a *Agent) runCommandSet(ctx context.Context, tc *taskContext, commandInfo
 			return errors.Wrap(ctx.Err(), "Agent stopped early")
 		}
 		tc.logger.Task().Infof("Finished %s in %s", fullCommandName, time.Since(start).String())
+		if a.endTaskResp != nil {
```

Comment:
Would it make sense to reuse the context cancellation machinery? Above on 141 we check for a cancelled context for timeouts. Since we know timeouts work correctly, it might make sense to do something similar for the task status case.

One question that the ticket doesn't answer though is whether the task should cease execution immediately if a status is set. We may wish to clarify that with the users first. I can imagine reasons that it should cease immediately (we know the status, so we don't need to keep going) and that it shouldn't (there's some addition error handling stuff the task wants to do).

Reply:
I think that it makes sense for the task to cease execution immediately unless it's a pre/post task; we want to simulate how tasks affect statuses. I added logic to ensure this behavior and added a should_continue option if the user does want to override that logic.